bundles/org.eclipse.ui.ide/src/org/eclipse/ui/views/markers/internal/ContentGeneratorDescriptor.java

```diff
@@ -154,6 +155,13 @@ public IConfigurationElement[] getFilterReferences() {
 		return filterGroups;
 	}
 
+	/**
+	 * @return the IConfigurationElement for scope area configuration.
+	 */
+	public IConfigurationElement getScopeAreaConfigurationElement() {
+		IConfigurationElement[] childElements = configurationElement.getChildren(MARKER_SCOPE_AREA_CONFIGURATION);
+		return childElements.length > 0 ? childElements[0] : null;
```

Comment:
If I see it right, this would silently peek some random contribution, in case multiple exists. Surely nothing we want. 

This "extension" looks more like a product customization and not like a typical extension point. Have you considered to use preferences service & product customization instead of introducing extension point that can only be extended by a single bundle?

Reply:
There shouldn't be problems with multiple occurrences.
a specific markerContentGenerator is given in the constructor of MarkerSupportView.
markerScopeAreaConfiguration that I added is optionally part of markerContentGenerator (0 or 1 occurrences).
No change in behavior for existing clients if markerScopeAreaConfiguration not present.

I haven't looked at preferences service & product customization but this extension point was already there. 
I just added a small piece for configuring the scope area.

Reply:
Are there any other issues or can this be merged now ?